magento/magento2
# Move pieces of Magento_Review out of Magento_Customer

### Description (*)

This moves the last few pieces of review functionality out of Magento_Customer.

**Please note**, this causes a breaking change to Magento_Customer, and adds an `@api` annotated class to Magento_Review.  I think it's too late for 2.3 and probably has to be part of 2.4

### Fixed Issues (if relevant)

1. magento/magento2#18302: Unable to disable module Magento_Review

### Manual testing scenarios (*)

1. Log in to the admin panel
2. View any customer record
3. Click the "Product Reviews" tab
4. Ensure a grid appears

### Contribution checklist (*)
